### PR TITLE
Created generator class for validator tests

### PR DESCRIPTION
**Pull request by Anna-Lena-DLR**
- validator tests are generated when creating a new concept
- validate() is tested

### CODE REVIEW

**Comment by SaMuellerDLR** [comment]
@Anna-Lena-DLR Der Test generator muss auch getestet sein :)

**Comment by franzTobiasDLR** [comment]
I think the new test class should also be added to the AllTestsGen in a concept's test project... Otherwise it is not exectued in the test suite...

File: de.dlr.sc.virsat.model.concept.test/resources/expectedOutputFilesForGenerators/StructuralElementInstanceValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * This class is generated once, do your changes here
+ * 
+ * 
+ *
+ */
+	public class StructuralElementInstanceValidatorTest extends AStructuralElementInstanceValidatorTest {
+
+	@Test	
+	public void testStructuralElementInstanceValidator() { 
+	StructuralElementInstanceValidator validator = new StructuralElementInstanceValidator();
```
Comment (franzTobiasDLR) [inline de.dlr.sc.virsat.model.concept.test/resources/expectedOutputFilesForGenerators/StructuralElementInstanceValidatorTest.java]: Could we improve the indentation here? Content of a method should be indented ;)

File: de.dlr.sc.virsat.model.concept.test/src/de/dlr/sc/virsat/model/concept/generator/tests/GenerateValidatorTestsTest.xtend [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    def void testCreateForCategoryTests() {
+    	concept = '''
+			Concept «TEST_CONCEPT_NAME» {
+			}
+		'''.parse
+
+    	val concreteClassContents = generateValidatorTests.createConcreteClass(concept, concept)
+    	val abstractClassContents = generateValidatorTests.createAbstractClass(concept, concept)
+    	
+		GeneratorJunitAssert.assertEqualContent(concreteClassContents, "/resources/expectedOutputFilesForGenerators/StructuralElementInstanceValidatorTest.java")
```
Comment (franzTobiasDLR) [inline de.dlr.sc.virsat.model.concept.test/src/de/dlr/sc/virsat/model/concept/generator/tests/GenerateValidatorTestsTest.xtend]: I think here we could also improve the indentation?

File: de.dlr.sc.virsat.model.concept.test/src/de/dlr/sc/virsat/model/concept/generator/tests/GenerateValidatorTestsTest.xtend [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		val abstractClassfileName = generateValidatorTests.createAbstractClassFileName(concept, concept)
+		val expectedConcreteClassFileName = "../../testConcept.test/src/testConcept/validator/StructuralElementInstanceValidatorTest.java"
+		val expectedAbstractClassFileName = "../../testConcept.test/src-gen/testConcept/validator/AStructuralElementInstanceValidatorTest.java"
+
+		Assert.assertEquals("Concrete file name for the generated create add command is correct", expectedConcreteClassFileName, concreteClassfileName)
+		Assert.assertEquals("Concrete file name for the generated create add command is correct", expectedAbstractClassFileName, abstractClassfileName)
+	}
+	
+	@Test
+	def void testCreateForCategoryTests() {
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.concept.test/src/de/dlr/sc/virsat/model/concept/generator/tests/GenerateValidatorTestsTest.xtend]: I think the test names are a bit strange. Maybe testCreateClass. I dont see what this test case has to do with categories.

**Pull request merged**